### PR TITLE
Extend profile completion by checking if any organization is linked

### PR DESCRIPTION
**Pull request by jakub-lotko**

### CODE REVIEW

File: src/HE.InvestmentLoans.WWW/Controllers/HomeController.cs [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
         }
-        else
+
+        if (!await _userContext.IsLinkedWithOrganization())
         {
-            return RedirectToAction("WhatTheHomeBuildingFundIs", "Guidance");
+            return RedirectToAction(nameof(OrganizationController.SearchOrganization), new ControllerName(nameof(OrganizationController)).WithoutPrefix());
         }
+
+        return RedirectToAction("Dashboard", "Home");
```
Comment (dawid-pawulski) [inline src/HE.InvestmentLoans.WWW/Controllers/HomeController.cs:37]: After logout user is redirect to home page. And regarding if user IsAuthenticated is taken do dashboard or guidance. 
So now we have to merge all solutions and leave also this redirect RedirectToAction("WhatTheHomeBuildingFundIs", "Guidance");

**Review by dawid-pawulski — CHANGES_REQUESTED** [review]

**Review by dawid-pawulski — APPROVED** [review]

File: src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs [hunk 2]
```diff
@@ -80,6 +80,15 @@ public async Task<bool> IsProfileCompleted()
         return userDetails.IsProfileCompleted();
     }
 
+    public async Task<bool> IsLinkedWithOrganization()
+    {
+        var userAccount = await _cacheService.GetValueAsync(
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/User/LoanUserContext.cs:85]: Method `LoadUserAccount` could set a new property `IsLinkedWithOrganisation`?

File: src/HE.InvestmentLoans.BusinessLogic/User/Repositories/LoanUserRepository.cs [hunk 3]
```diff
@@ -24,12 +25,17 @@ public LoanUserRepository(IOrganizationServiceAsync2 serviceClient, IContactServ
...
     public async Task<ContactRolesDto?> GetUserAccount(UserGlobalId userGlobalId, string userEmail)
     {
-        var contactRoles = await _contactService.GetContactRoles(_serviceClient, userEmail, PortalConstants.PortalType, userGlobalId.ToString())
-                            ?? throw new NotFoundException("Contact role", userGlobalId);
+        var contactRoles = await _contactService.GetContactRoles(_serviceClient, userEmail, PortalConstants.PortalType, userGlobalId.ToString());
 
         return contactRoles;
     }
 
+    public Task LinkAccountToOrganization(UserGlobalId userGlobalId, CompaniesHouseNumber organizationNumber)
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.BusinessLogic/User/Repositories/LoanUserRepository.cs:33]: Rename to `LinkContactToOrganisation`

File: src/HE.InvestmentLoans.Contract/Organization/LinkAccountWithOrganization.cs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+using HE.InvestmentLoans.Contract.Organization.ValueObjects;
+using MediatR;
+
+namespace HE.InvestmentLoans.Contract.Organization;
+public record LinkAccountWithOrganization(CompaniesHouseNumber Number) : IRequest;
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.Contract/Organization/LinkAccountWithOrganization.cs:5]: LinkAccountWithOrganization -> LinkAccountWithOrganizationCommand etc

File: src/HE.InvestmentLoans.Contract/Organization/ValueObjects/CompaniesHouseNumber.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+using HE.InvestmentLoans.Contract.Domain;
+
+namespace HE.InvestmentLoans.Contract.Organization.ValueObjects;
+public class CompaniesHouseNumber : ValueObject
+{
+    private readonly string _value;
+
+    public CompaniesHouseNumber(string value)
+    {
+        _value = value;
```
Comment (lukasz-ciura-pwc) [inline src/HE.InvestmentLoans.Contract/Organization/ValueObjects/CompaniesHouseNumber.cs:10]: Add validation for empty strings

**Review by lukasz-ciura-pwc — APPROVED** [review]

**Pull request merged**